pynxtools/dataconverter/helpers.py

```diff
@@ -629,7 +627,7 @@ def validate_data_dict(template, data, nxdl_root: ET.Element):
                 data[path] = is_valid_data_field(data[path], nxdl_type, path)
                 is_valid_enum, enums = is_value_valid_element_of_enum(data[path], elem)
```

Comment:
This seems to check only the given nxdl elem which concept the data element directly belongs to. Note that the enum list may have been inherited. In case of enums, one should check the first elem in the returned elist which contains an enum list rather than checking the absolut first elem.